src/App.js

```diff
@@ -0,0 +1,129 @@
+import styled, { ThemeProvider, createGlobalStyle, css } from 'styled-components/macro';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import { Transition, TransitionGroup } from 'react-transition-group';
+import { Helmet, HelmetProvider } from 'react-helmet-async';
+import Header from './components/Header';
+import Theme from './utils/Theme';
+
+const Home = lazy(() => import('./screens/Home'));
+const NotFound = lazy(() => import('./screens/NotFound'));
+const Changelog = lazy(() => import('./screens/Changelog'));
+const Donate = lazy(() => import('./screens/Donate'));
+
+export const AppContext = createContext();
+
+function App() {
+  useEffect(() => {
+    window.history.scrollRestoration = 'manual';
+  }, []);
+
+  return (
+    <HelmetProvider>
+      <ThemeProvider theme={Theme}>
+        <React.Fragment>
+          <Helmet>
+            <title>Impact Client - A Minecraft Utility Mod</title>
+            <meta
+              name="description"
+              content="Impact Client is an advanced utility mod for Minecraft, it is based on ClientAPI and includes a large number of useful mods."
+            />
+          </Helmet>
+          <GlobalStyles />
+          <Router>
+            <Route render={({ location }) => (
+              <React.Fragment>
+                <Transition
+                  mountOnEnter
+                  unmountOnExit
+                  in={location.pathname !== '/'}
+                  timeout={{enter: 0, exit: 400}}
+                >
+                  {status => <Header status={status} />}
+                </Transition>
+                <TransitionGroup component={React.Fragment}>
+                  <Transition key={location.pathname} timeout={300}>
+                    {status => (
+                      <MainContent status={status} id="MainContent" role="main" tabIndex={-1}>
+                        <Helmet>
+                          <link rel="canonical" href={`https://impactdevelopment.github.io${location.pathname}`} />
+                        </Helmet>
+                        <AppContext.Provider value={{ status }}>
+                          <Suspense fallback={<React.Fragment />}>
+                            <Switch location={location}>
+                              <Route exact path="/" component={Home} />
+                              <Route path="/donate" component={Donate} />
+                              <Route path="/changelog" component={Changelog} />
+                              <Route component={NotFound} />
+                            </Switch>
+                          </Suspense>
+                        </AppContext.Provider>
+                      </MainContent>
+                    )}
+                  </Transition>
+                </TransitionGroup>
+              </React.Fragment>
+            )} />
+          </Router>
+        </React.Fragment>
+      </ThemeProvider>
+    </HelmetProvider>
+  );
+};
```

Comment:
JSX is cool and awesome, however if we were to go down this route gh-pages wouldn't be the place to do it. IMO react should always be rendered server-side and simply add to the experience client-side. That'd likely require paying for a service such as heroku to host the site (depending how much traffic we get @leijurv ?).

Seems overkill for what only needs to be a simple website with a download link.

Reply:
This was merely an exercise that I wanted to share to get feedback.
However, if you are looking for help, I can work with your team to design a website from the ground-up as well as build it. It doesn't have to be built with React; I can make static websites all-the-same. I am looking to sharpen my skills without any cost to your team. I found Impact Development through your lead developer in previous projects.

Anyways, I have a server you can use if hosting is an issue.
Alternatively, there are services like [Netlify](https://www.netlify.com/) which are free to use.
Netlify, in particular, builds from your stable branch, keeping your site up-to-date.
You can work with these tools from your browser with [CodeSandbox](https://codesandbox.io/), which will take care of Github and deployment.

With my server, I can also hook you up with my Cloudflare subscription which makes the site much faster and still gives you access to SSL. For reference, all of my works have scored in the top 95% of websites in speed and accessibility thanks to this CDN.